apis/risingwave/ng/v1alpha1/risingwave_types.go

```diff
@@ -0,0 +1,250 @@
+	ImagePullSecrets []string                    `json:"imagePullSecrets,omitempty"`
+	UpgradeStrategy  appsv1.DeploymentStrategy   `json:"upgradeStrategy,omitempty"`
+	Resources        corev1.ResourceRequirements `json:"resources,omitempty"`
+	NodeSelector     map[string]string           `json:"nodeSelector,omitempty"`
+	PodTemplate      *string                     `json:"podTemplate,omitempty"`
+}
+
+type RisingWaveComponentGroup struct {
+	Name      string                            `json:"name,omitempty"`
+	Replicas  int64                             `json:"replicas,omitempty"`
+	RestartAt *metav1.Time                      `json:"restartAt,omitempty"`
+	Template  *RisingWaveComponentGroupTemplate `json:",inline"`
+}
+
+type RisingWaveComputeGroupTemplate struct {
+	RisingWaveComponentGroupTemplate `json:",inline"`
+	VolumeMounts                     []corev1.VolumeMount `json:"volumeMounts,omitempty"`
+}
+
+type RisingWaveComputeGroup struct {
+	Name      string                          `json:"name,omitempty"`
+	Replicas  int64                           `json:"replicas,omitempty"`
+	RestartAt *metav1.Time                    `json:"restartAt,omitempty"`
+	Template  *RisingWaveComputeGroupTemplate `json:",inline"`
+}
+
+type RisingWaveComponentMetaPorts struct {
+	RisingWaveComponentCommonPorts `json:",inline"`
+	Dashboard                      int32 `json:"dashboard,omitempty"`
+}
+
+type RisingWaveComponentMeta struct {
+	Ports  RisingWaveComponentMetaPorts `json:"ports,omitempty"`
+	Groups []RisingWaveComponentGroup   `json:"groups,omitempty"`
+}
+
+type RisingWaveComponentCommonPorts struct {
+	Service int32 `json:"service,omitempty"`
+	Metrics int32 `json:"metrics,omitempty"`
+}
+
+type RisingWaveComponentFrontend struct {
+	Ports  RisingWaveComponentCommonPorts `json:"ports,omitempty"`
+	Groups []RisingWaveComponentGroup     `json:"groups,omitempty"`
+}
+
+type RisingWaveComponentCompute struct {
+	Ports  RisingWaveComponentCommonPorts `json:"ports,omitempty"`
+	Groups []RisingWaveComputeGroup       `json:"groups,omitempty"`
+}
+
+type RisingWaveComponentCompactor struct {
+	Ports  RisingWaveComponentCommonPorts `json:"ports,omitempty"`
+	Groups []RisingWaveComponentGroup     `json:"groups,omitempty"`
+}
+
+type RisingWaveComponentsSpec struct {
+	Meta      RisingWaveComponentMeta      `json:"meta,omitempty"`
+	Frontend  RisingWaveComponentFrontend  `json:"frontend,omitempty"`
+	Compute   RisingWaveComponentCompute   `json:"compute,omitempty"`
+	Compactor RisingWaveComponentCompactor `json:"compactor,omitempty"`
+}
+
+type RisingWaveMetaStorage struct {
+	Memory *bool   `json:"memory,omitempty"`
+	Etcd   *string `json:"etcd,omitempty"`
+}
+
+type RisingWaveObjectStorageMinIO struct {
+	Endpoint string `json:"endpoint,omitempty"`
+	Bucket   string `json:"bucket,omitempty"`
+}
+
+type RisingWaveObjectStorageS3 struct {
+	Secret string `json:"secret,omitempty"`
+}
+
+type RisingWaveObjectStorage struct {
```

Comment:
We will have PVs and S3/MinIO at the same time after RisingWave supports spilling and secondary cache.

Reply:
Right, that's why I design the interfaces `pvcTemplates` and `volumnMounts` for the compute nodes. You can check it in the `risingwave_ng.yaml` file.